### PR TITLE
Remove pyyaml as a dependency of Kedro-Telemetry

### PR DESCRIPTION
**Pull request by datajoely**
## Description
Mirror of this PR on core https://github.com/kedro-org/kedro/issues/1541

## Development notes
- See above
- I think I messed up my git history so have two commits in here, sorry 😓 ! 

## Checklist

- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [x] Added a description of this change in the relevant `RELEASE.md` file
- [x] Added tests to cover my changes

### CODE REVIEW

**Comment by MerelTheisenQB** [comment]
I'm not sure you meant to push the changes to the docs?

**Comment by datajoely** [comment]
@MerelTheisenQB I reverted two commits, not one (I didn't realise I was on main) so the doc change went in previously 🙈

File: kedro-telemetry/RELEASE.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+# Release 0.2.2
+
+## Bug fixes and other changes
+* Removed explicit `PyYAML` requirement since it is already a dependency of `Kedro`
```
Comment (MerelTheisenQB) [inline kedro-telemetry/RELEASE.md:4]: We haven't released `0.2.1` so this should go in the section below.

File: kedro-telemetry/RELEASE.md [hunk 2]
```diff
@@ -1,3 +1,8 @@
...
+
 # Release 0.2.1
```
Comment (MerelTheisenQB) [inline kedro-telemetry/RELEASE.md:6]: To make it clear we haven't actually released this yet!

```suggestion
# Upcoming release 0.2.1
```

**Review by MerelTheisenQB — APPROVED** [review]
👍

**Comment by datajoely** [comment]
Okay addressed questions - are the e2e tests a known issue?

**Review by AntonyMilneQB — APPROVED** [review]

**Comment by AntonyMilneQB** [comment]
> Okay addressed questions - are the e2e tests a known issue?

Hmmm no I don't think so. Is it passing on `main` or does it give the same error?

**Comment by datajoely** [comment]
@AntonyMilneQB so the I think this is the first time `kedro-airflow` code has changed in a while, so I'm not sure the CI has run recently

File: kedro-docker/project_dummy/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,23 @@
+ARG BASE_IMAGE=python:3.6-buster
```
Comment (MerelTheisenQB) [inline kedro-docker/project_dummy/Dockerfile:1]: Do we need this `Dockerfile`?

**Review by MerelTheisenQB — APPROVED** [review]
Great investigation @noklam ! 🕵️

File: kedro-docker/project_dummy/Dockerfile (hunk 3 above)
Comment (noklam) [reply kedro-docker/project_dummy/Dockerfile]: No we don't!

**Pull request merged**